### DISCUSSION TITLE
Firebase Identity Platform TOTP (with Authenticator) availability in Android/iOS SDKs

### DISCUSSION THREAD

**Discussion by billypap1** [firebase/firebase-ios-sdk#11390] [Q&A]
I want to upgrade my Firebase project in order to use the Identity Platform. As I read the in docs there is now the option to enable MFA with TOTP (Authenticator) except from SMS. According to these docs-> [https://cloud.google.com/identity-platform/docs/admin/enabling-totp-mfa](https://cloud.google.com/identity-platform/docs/admin/enabling-totp-mfa) you can enable the TOTP for web apps but I can't find any documentation about integrating TOTP with Authenticator to mobile apps.

Is TOTP with Authenticator available for iOS/Android apps SDKs?

If it's not available for Android/iOS SDKs, a user who has enabled MFA TOTP in the web app, can't log in in mobile sdks.

**Comment by paulb777** [comment]
The iOS SDK support for TOTP is in active implementation. There have been several recent commits to the branch https://github.com/firebase/firebase-ios-sdk/commits/totp-release-0413

**Reply by billypap1** [reply]
@paulb777  Will it be available in both SDKs (Android - iOS) soon? Is there an ETA?
Because right now, it is available in web sdk only, so we can't use TOTP via authenticator at all in our apps (Web - Android - iOS).
